add option for srcAddress and srcPort

Links2004 opened the pull request:
add option for srcAddress and srcPort.
needed if you have an server with multiple interfaces
for example LAN + WAN then you dont want to send the WOL msg to over the WAN interface

Links2004 committed:
add option for srcAddress and srcPort

agnat commented:
Setting a source address makes sense.

However, IMO setting a port doesn't. On the contrary. By letting the OS pick a port the old code avoided all kinds of trouble (port busy, &c.). The new code randomly fails whenever your port randomization has bad luck.

Also, why the odd port range?

Links2004 commented:
you are right the port selection can fail.

The port range has no rail meaning, i only wanted to avoid the lower ranges, in use here(10000 - 60000).
all < 1024 is root only.
and < 10000 has an higher change to be already in use.

for socket.bind port is required, so i have no possibility to only set the address.

socket.bind(port[, address][, callback])
https://nodejs.org/api/dgram.html#dgram_socket_bind_port_address_callback

is there a better method to bind the udp socked to an specific interface?

agnat commented:
IIRC using port 0 is what you want. But this is your contribution. So, look it up... ;)

Links2004 committed:
use 0 as srcPort default

0 = random src port selected by system (not in nodejs documentation)
verified with wireshark

Links2004 commented:
0 as port is working fin.
i have verified with Wireshark.

agnat commented:
Yeah, well. You verified it on one platform. However, [Unix Network Programming](http://www.unixnetworkprogramming.com) says I'm right... which is good enough for me.

Now, why keep the port option at all? As I said: I can not see any use case. It only serves to shoot your own foot by catching a busy port. Please remove it.

Links2004 committed:
remove srcPort port option

Links2004 commented:
is removed

agnat commented on wake_on_lan.js:
```diff
@@ -43,6 +43,7 @@ exports.wake = function(mac, opts, callback) {
     , num_packets = opts['num_packets'] || 3
     , interval    = opts['interval']    || 100
     , port        = opts['port']        || 9
+	, srcAddress  = opts['srcAddress']  || '0.0.0.0'
```
Indentation: Is that a tab?

Links2004 replied:
yes

Links2004 committed:
code style

agnat commented on package.json:
```diff
@@ -21,6 +21,9 @@
     { "name":   "Jann Horn"
     , "github": "thejh"
+    }, 
+	{ "name":   "Markus Sattler"
```
Indentation: Looks like another tab. Please use grep or something to look for others...
